### PR TITLE
Fixing unreachable method overloads

### PR DESCRIPTION
**Pull request by erik-krogh**
Sometimes developers write method overloads that cannot be used.  
For example:

```TypeScript
class ContentBlock extends Record {
   ...
   getType(): DraftBlockType;
   getType(): string;
}
```

When a client invokes an overloaded method [the first matching overload will always be used](https://github.com/microsoft/TypeScript/blob/master/doc/spec.md#4.15.1), and hence the following program will result in a type-error: 

```TypeScript
var foo: string = new ContentBlock().getType();
```

The second method overload is therefore unreachable, and the overload will cause nothing but confusion for developers writing a client or maintaining the library.

Often the mistake is caused by a simple duplicate method, which makes no difference, but can be confusing.
Other times the pattern is mistakenly used instead of having a union-type as the return. 
And a last reason is more tricky, consider the following: 

```TypeScript
interface ShallowRenderer {
    getRenderOutput<E extends ReactElement>(): E;
    getRenderOutput(): ReactElement;
} 
```

The first overload of the above program will always be used when writing a client using the library. This happens even if no type-parameters are used, as the TypeScript compiler will simply insert a default type parameter. 

This pull request attempts to fix all trivially discoverable instances of this mistake. 

Please fill in this template.

- [x] Use a meaningful title for the pull request. ~~Include the name of the package modified.~~
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

~~If changing an existing definition:~~
- [ ] ~~Provide a URL to documentation or source code which provides context for the suggested changes:~~
- [ ] ~~If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.~~
- [ ] ~~If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.~~

### CODE REVIEW

**Review by aj-r — CHANGES_REQUESTED** [review]

**Review by SimonSchick — APPROVED** [review]

**Review by MunifTanjim — APPROVED** [review]
Changes for `@types/draft-js` looks alright! :+1:

File: types/lodash/common/array.d.ts [hunk 1]
```diff
@@ -984,32 +984,32 @@ declare module "../index" {
...
 
         /**
          @see _.fromPairs
          */
-        fromPairs(
-            this: LoDashImplicitWrapper<List<any[]> | null | undefined>
-        ): LoDashImplicitWrapper<Dictionary<any>>;
+        fromPairs<T>(
+            this: LoDashImplicitWrapper<List<[PropertyName, T]> | null | undefined>
+          ): LoDashImplicitWrapper<Dictionary<T>>;
```
[the remark this answers is not in the document]
Comment (erik-krogh) [reply types/lodash/common/array.d.ts]: Yeah, you're right. 
Both possible orderings of the overloads are problematic though.   
Here I've showcased the problem with the old ordering: https://bit.ly/31YyVlD 

But yes. The old approach was likely the best, so I'll go ahead and revert this change.

**Review by jgoz — APPROVED** [review]
`react-test-renderer` 👍

**Review by aj-r — APPROVED** [review]

**Review by andrewbranch — APPROVED** [review]
This is great, thank you for doing this!

**Pull request merged**